dask_cuda/tests/test_dask_cuda_worker.py

```diff
@@ -294,6 +294,28 @@ def test_cuda_visible_devices_uuid(loop):  # noqa: F811
                     assert list(result.values())[0] == gpu_uuid
 
 
+@patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0"})
+def test_no_nanny(loop):  # noqa: F811
+    with popen(["dask-scheduler", "--port", "9069", "--no-dashboard"]):
+        with popen(
+            [
+                "dask-cuda-worker",
+                "127.0.0.1:9069",
+                "--host",
+                "127.0.0.1",
+                "--no-dashboard",
+                "--no-nanny",
+            ],
+            flush_output=False,
+        ) as worker:
+            with Client("127.0.0.1:9069", loop=loop) as client:
+                assert wait_workers(client, n_gpus=1)
+                assert any(
+                    b"distributed.nanny" not in worker.stdout.readline()
+                    for i in range(5)
```

Comment:
What does the magic number `5` mean here?

Reply:
This is to capture the first few lines and confirm nanny doesn't show up. For example, with nannies:

```
2022-06-14 05:59:22,053 - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.162:36881'
2022-06-14 05:59:24,289 - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
2022-06-14 05:59:25,043 - distributed.preloading - INFO - Run preload setup click command: dask_cuda.initialize
2022-06-14 05:59:25,043 - distributed.worker - INFO -       Start worker at:  tcp://10.33.227.162:33455
2022-06-14 05:59:25,044 - distributed.worker - INFO -          Listening to:  tcp://10.33.227.162:33455
```

without nannies:

```
(dask-dev) bzaitlen@dgx12:/datasets/bzaitlen/GitRepos/dask-cuda$ CUDA_VISIBLE_DEVICES=0 dask-cuda-worker tcp://10.33.227.162:8786 --no-nanny
2022-06-14 05:59:47,314 - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
2022-06-14 05:59:47,960 - distributed.preloading - INFO - Run preload setup click command: dask_cuda.initialize
2022-06-14 05:59:47,961 - distributed.worker - INFO -       Start worker at:  tcp://10.33.227.162:42383
2022-06-14 05:59:47,961 - distributed.worker - INFO -          Listening to:  tcp://10.33.227.162:42383
```

I can add a comment to explain this

Reply:
Why can't we go through all the lines? This seems like something that may change with time and the 5 lines won't anymore capture what we want to see (i.e., test will always pass even if there's